test/fakeintake/server/server.go

```diff
@@ -247,16 +249,37 @@ func (fi *Server) handleGetPayloads(w http.ResponseWriter, req *http.Request) {
 		})
 		return
 	}
+
 	// we could support multiple endpoints in the future
 	route := routes[0]
-	log.Printf("Handling GetPayload request for %s payloads", route)
-	payloads := fi.safeGetPayloads(route)
-
-	// build response
-	resp := api.APIFakeIntakePayloadsGETResponse{
-		Payloads: payloads,
+	log.Printf("Handling GetPayload request for %s payloads.", route)
+	var jsonResp []byte
+	var err error
+	if req.URL.Query().Get("format") == "" {
+		payloads := fi.safeGetRawPayloads(route)
+
+		// build response
+		resp := api.APIFakeIntakePayloadsRawGETResponse{
+			Payloads: payloads,
+		}
+		jsonResp, err = json.Marshal(resp)
+	} else if fi.payloadParser.isValidRoute(route) {
+		payloads, _ := fi.safeGetJsonPayloads(route)
```

Comment:
Is there a reason to ignore the error?

Reply:
``safeGetJsonPayloads`` only returns an error when  ``route`` is Invalid or ``route`` is valid but no data exist yet( ``payload, err := fi.payloadParser.getJsonPayload(route)``). Since we check that the route is valid before hand, if an error is outputed, we know it's only because it doesn't have any data yet. In this case we don't want to output any error, we'll just output an empty payload data ``{Payloads:{}}``

Reply:
Based on https://github.com/DataDog/datadog-agent/pull/18332/files#diff-067626d001d8197c8320bdd344daa17d8459b1d48eeecc14bc74ce50877bedcdR30-R36, `getJsonPayload` returns an error only when the route is not supported.
Anyways, it is better to not assume the kind of error returned by a function as it may evolve over the time so I think this error should be handle properly.

Reply:
I use ``safeGetJsonPayloads`` not ``getJsonPayload``

Reply:
For this very specific scenario, the only possible error is that a route isn't supported but to access this code ``fi.payloadParser.IsRouteHandled(route)`` we check before hand that the route is supported, so I still think that it will just create dead code and that if the function is changed (adding a new error) then it's normal to adjust code where they're called. But you convinced me, it's safer I'll handle this error as well.